RiotSwiftUI/Modules/Template/SimpleUserProfileExample/Model/TemplateUserProfileStateAction.swift

```diff
@@ -17,5 +17,7 @@
 import Foundation
 
 enum TemplateUserProfileStateAction {
+    case incrementCount
```

Comment:
I wonder if we should try to categorise state actions by type (user action, service action)
There would be something like:
```
case viewAction(TemplateUserProfileViewAction)
```

It would avoid to duplicate things. The view model will be aware of all user actions. `done` is probably useless but `cancel` can be useful to cancel any pending request.

Reply:
Yea we can do that for view actions that go directly to state modification we can pass on the view event.

Not sure I fully understand the second bit. Do you mean in the reducer? We should only be doing synchronous state modifications in the reducer.

If we are doing any work in the service that has a subscription stored in `cancellables` they will all be cancelled on `deinit` of the viewmodel.
But yea if there is any other manual cleanup we needed to do we could use `cancel` action in `process(viewAction:)`

Reply:
Updated with the first suggestion you made.